backend/application/zfs/pool_management_service.py

```
@@ -0,0 +1,432 @@
+                'has_errors': pool.has_errors(),
+                'needs_attention': pool.needs_attention(),
+                'dataset_count': len(pool.datasets)
+            })
+            
+            return status
+            
+        except (ZFSOperationError, ZFSPoolNotFoundError):
+            raise
+        except Exception as e:
+            logger.error(f"Failed to get pool status for {pool_name}: {e}")
+            raise ZFSOperationError(f"Failed to get pool status for {pool_name}: {e}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)